### ISSUE TITLE
ToNotifyCollectionChanged: Collection.Clear() causes InvalidOperationException when binding to ListBox(WPF)

### ISSUE THREAD

**Issue by vabock** [Cysharp/ObservableCollections#28]
The following code will cause an `InvalidOperationException` when `ObservableList.Clear()` is executed.

```xml
<Window x:Class="WpfApp1.MainWindow" ...>
    <StackPanel>
        <ListBox ItemsSource="{Binding ItemsView}" />
        <Button Content="Clear" Command="{Binding ClearCommand}" />
    </StackPanel>
</Window>
```

```c#
using ObservableCollections;
using R3;
using System.Windows.Data;

namespace WpfApp1
{
    class ViewModel
    {
        private ObservableList<int> observableList { get; } = new ObservableList<int>();
        public INotifyCollectionChangedSynchronizedView<int> ItemsView { get; }
        public ReactiveCommand<Unit> ClearCommand { get; } = new ReactiveCommand<Unit>();
        
        public ViewModel()
        {
            observableList.Add(1);
            observableList.Add(2);

            ItemsView = observableList.CreateView(x => x).ToNotifyCollectionChanged();

            BindingOperations.EnableCollectionSynchronization(ItemsView, new object());

            ClearCommand.Subscribe(_ => observableList.Clear());
        }
    }
}
```

```
R3 UnhandledException:System.InvalidOperationException: Collection Remove event must specify item position.
   at MS.Internal.Data.EnumerableCollectionView.ProcessCollectionChanged(NotifyCollectionChangedEventArgs args)
   at System.Windows.Data.CollectionView.PostChange(NotifyCollectionChangedEventArgs args)
   at ObservableCollections.Internal.NotifyCollectionChangedSynchronizedView`2.OnCollectionChanged(ChangedKind changedKind, T value, TView view, NotifyCollectionChangedEventArgs`1& eventArgs) in C:\Source\ObservableCollections-2.0.2\src\ObservableCollections\Internal\NotifyCollectionChangedSynchronizedView.cs:line 72
   at ObservableCollections.Internal.NotifyCollectionChangedSynchronizedView`2.ObservableCollections.ISynchronizedViewFilter<T,TView>.OnCollectionChanged(ChangedKind changedKind, T value, TView view, NotifyCollectionChangedEventArgs`1& eventArgs)
   at ObservableCollections.SynchronizedViewFilterExtensions.InvokeOnRemove[T,TView](ISynchronizedViewFilter`2 filter, T value, TView view, NotifyCollectionChangedEventArgs`1& eventArgs) in C:\Source\ObservableCollections-2.0.2\src\ObservableCollections\ISynchronizedViewFilter.cs:line 96
   at ObservableCollections.SynchronizedViewFilterExtensions.InvokeOnRemove[T,TView](ISynchronizedViewFilter`2 filter, ValueTuple`2 value, NotifyCollectionChangedEventArgs`1& eventArgs) in C:\Source\ObservableCollections-2.0.2\src\ObservableCollections\ISynchronizedViewFilter.cs:line 91
   at ObservableCollections.ObservableList`1.View`1.SourceCollectionChanged(NotifyCollectionChangedEventArgs`1& e) in C:\Source\ObservableCollections-2.0.2\src\ObservableCollections\ObservableList.Views.cs:line 237
   at ObservableCollections.ObservableList`1.Clear() in C:\Source\ObservableCollections-2.0.2\src\ObservableCollections\ObservableList.cs:line 118
   at WpfApp1.ViewModel.<.ctor>b__9_1(Unit _) in C:\Source\WpfApp1\ViewModel.cs:line 22
   at R3.AnonymousObserver`1.OnNextCore(T value)
   at R3.Observer`1.OnNext(T value)
```

It occurs because `eventArgs.OldStartingIndex` is `-1`, which is created by `NotifyCollectionChangedEventArgs<T>.Reset()`.

https://github.com/Cysharp/ObservableCollections/blob/bdbb5c0c76a1b7677a7107b65e2f94235e98429d/src/ObservableCollections/Internal/NotifyCollectionChangedSynchronizedView.cs#L71-L74

https://github.com/Cysharp/ObservableCollections/blob/bdbb5c0c76a1b7677a7107b65e2f94235e98429d/src/ObservableCollections/ObservableList.Views.cs#L232-L241

I tried fixing ObservableList.View, but could not figure out how to fix ObservableDictionary, ObservableHashSet... that does not have Index.

https://github.com/vabock/ObservableCollections/commit/13c01400991eaf1f4dfe06f8b38c49a53197b38d

dotnet/efcore#7397 Same problem encountered with non-indexed collections and ListBox.

**Comment by neuecc** [comment]
thanks, simply I'll add ChangedKind.Clear that notify NotifyCollectionChangedAction.Reset.

**Comment by neuecc** [comment]
We've released 2.1.0,
A breaking change was made regarding SynchronizedView to more correctly communicate INotifyCollectionChanged.
